### PR TITLE
feature: Disable the input field if the last request has failed

### PR DESCRIPTION
**Pull request by denys-kolomiitsev**
#350

### CODE REVIEW

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Deployment status: success

File: src/components/Chat/SendMessageButton.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { IconSend } from '@tabler/icons-react';
+
+import classNames from 'classnames';
+
+import { ConversationsSelectors } from '@/src/store/conversations/conversations.reducers';
+import { useAppSelector } from '@/src/store/hooks';
+import { ModelsSelectors } from '@/src/store/models/models.reducers';
+
+interface Props {
+  messageIsStreaming: boolean;
```
Comment (mikitabut) [inline src/components/Chat/SendMessageButton.tsx]: I think you can use just selector inside component

File: src/store/conversations/conversations.reducers.ts [hunk 2]
```diff
@@ -582,6 +582,26 @@ const selectIsMessagesError = createSelector(
...
+      const lastMessageIndex = conv.messages.length - 1;
+      const lastMessage = conv.messages[lastMessageIndex];
+      return (
+        lastMessage.role === 'assistant' && lastMessage.content.length === 0
+      );
+    });
+  },
+);
+
+const selectIsNotModelTypeModelInSelectedConversations = createSelector(
```
Comment (mikitabut) [inline src/store/conversations/conversations.reducers.ts]: maybe to rename to selectNotModelConversations?

File: src/store/conversations/conversations.reducers.ts [hunk 3]
```diff
@@ -582,6 +582,26 @@ const selectIsMessagesError = createSelector(
...
 );
 
+const selectIsLastAssistantMessageEmpty = createSelector(
+  [selectSelectedConversations],
+  (conversations) => {
+    return conversations.some((conv) => {
+      const lastMessageIndex = conv.messages.length - 1;
+      const lastMessage = conv.messages[lastMessageIndex];
+      return (
+        lastMessage.role === 'assistant' && lastMessage.content.length === 0
```
Comment (mikitabut) [inline src/store/conversations/conversations.reducers.ts:597]: What if you have no any messages in conversation?

**Review by mikitabut — CHANGES_REQUESTED** [review]

File: src/components/Chat/SendMessageButton.tsx (hunk 1 above)
Comment (denys-kolomiitsev) [reply src/components/Chat/SendMessageButton.tsx]: Good suggestion.
I've changed to use selector instead.

File: src/store/conversations/conversations.reducers.ts (hunk 2 above)
Comment (denys-kolomiitsev) [reply src/store/conversations/conversations.reducers.ts]: Thank you for suggestion.
Constant name have been changed to the selectNotModelConversations.

File: src/store/conversations/conversations.reducers.ts (hunk 3 above)
Comment (denys-kolomiitsev) [reply src/store/conversations/conversations.reducers.ts:597]: Condition added.
Thank you for pointing this out!

**Review by mikitabut — APPROVED** [review]

**Comment by mikitabut** [comment]
/deploy-review
>Deployment status: success

**Pull request merged**